doc/sphinx/proofs/writing-proofs/rewriting.rst

```diff
@@ -522,28 +505,26 @@ the conversion in hypotheses :n:`{+ @ident}`.
    information about the constants it encounters and the unfolding decisions it
    makes.
 
-.. tacn:: red
-   :name: red
+.. tacn:: red {? @occurrences }
 
-   This tactic applies to a goal that has the form::
+   :math:`\beta`:math:`\iota`:math:`\zeta`-reduces `T`, if possible, in the selected
+   hypotheses and/or the goal, which must have the form:
 
-     forall (x:T1) ... (xk:Tk), T
+     :n:`forall @open_binders, T`
```

Comment:
It's permitted to have no binders
```suggestion
     :n:`{? forall @open_binders, } T`
```

Reply:
Got it.  Is there any constraint on `T` that I can add?

Reply:
`T` is not headed by a `forall` (that's the point of describing it in this way, so you can talk about the head underneath the foralls)